eventsourcing/src/test/java/org/axonframework/eventsourcing/eventstore/StorageEngineTestSuite.java

```diff
@@ -173,25 +172,42 @@ void usingConsistencyMarkerFromSourcingEventToAppendAfterAppendWithConditionIsNo
     }
 
     @Test
-    void sourcingEventsReturnsEmptyStreamIfNoEventsInTheStoreForFlux() {
+    void sourcingEventsReturnsConsistencyMarkerAsSoleMessageWhenNoEventsInTheStoreForFlux() {
         SourcingCondition testCondition = SourcingCondition.conditionFor(TEST_CRITERIA);
 
         StepVerifier.create(testSubject.source(testCondition).asFlux())
-                    .expectNextCount(0)
+                    .assertNext(StorageEngineTestSuite::assertMarkerEntry)
                     .verifyComplete();
     }
 
     @Test
-    void sourcingEventsReturnsEmptyStreamThatCompletesIfNoEventsInTheStore() {
+    void sourcingEventsReturnsConsistencyMarkerAsSoleMessageAndCompletesWhenNoEventsInTheStore() {
         AtomicBoolean completed = new AtomicBoolean(false);
         SourcingCondition testCondition = SourcingCondition.conditionFor(TEST_CRITERIA);
 
-        testSubject.source(testCondition)
-                   .whenComplete(() -> completed.set(true))
-                   .first()
-                   .asCompletableFuture();
+        MessageStream<EventMessage<?>> sourcingStream = testSubject.source(testCondition)
+                                                                   .whenComplete(() -> completed.set(true));
+        await("Await first entry availability")
+                .pollDelay(Duration.ofMillis(50))
+                .atMost(Duration.ofMillis(500))
+                .until(sourcingStream::hasNextAvailable);
+        Optional<Entry<EventMessage<?>>> entry = sourcingStream.next();
+        assertTrue(entry.isPresent());
+        assertMarkerEntry(entry.get());
+        await("Await end of stream")
+                .pollDelay(Duration.ofMillis(50))
+                .atMost(Duration.ofMillis(500))
+                .until(() -> !sourcingStream.hasNextAvailable());
+
+        await("Awaiting until sourcing completes")
+                .pollDelay(Duration.ofMillis(50))
+                .atMost(Duration.ofMillis(500))
+                .untilTrue(completed);
+    }
 
-        await("Awaiting until sourcing completes").untilTrue(completed);
+    private static void assertMarkerEntry(Entry<EventMessage<?>> entry) {
+        assertNotNull(entry.getResource(ConsistencyMarker.RESOURCE_KEY));
```

Comment:
Shouldn't we assert the marker entry values in both this test and the `AggregateBasedStorageEngineTestSuite`?

Reply:
The test suite doesn't now what kind of `ConsistencyMarker` implementation is used. Hence why it doesn't check the position, as there's no 100% certainty it's a position at all times.
Then again, perhaps I can make a test-suite implementation specific assertion somehow. Let me take a look there.

Reply:
I think that, to support an exact `ConsistencyMarker` value assertion, I need to make abstract methods for each test case using the `assertMarkerEntry` separately. Do you want me to? :sweat_smile: 

By the way, to add to my previous comment. I would expect that this `StorageEngineTestSuite` will be used by Axon Server, InMemory, Mongo, and several RDBMS-versions.
Just as with the `TrackingToken`, I am assuming the `ConsistencyMarker` implementation and thus optionally how the position is defined, may differ.